solr/core/src/java/org/apache/solr/core/SchemaCodecFactory.java

```diff
@@ -127,7 +132,8 @@ public KnnVectorsFormat getKnnVectorsFormatForField(String field) {
               if (DenseVectorField.HNSW_ALGORITHM.equals(knnAlgorithm)) {
                 int maxConn = vectorType.getHnswMaxConn();
                 int beamWidth = vectorType.getHnswBeamWidth();
-                return new Lucene95HnswVectorsFormat(maxConn, beamWidth);
+                var delegate = new Lucene95HnswVectorsFormat(maxConn, beamWidth);
+                return new SolrDelegatingKnnVectorsFormat(delegate, vectorType.getDimension());
```

Comment:
2/3 ... but then yes we get to here but `Lucene95HnswVectorsFormat` being final about its `getMaxDimensions` is a problem ...